### PR TITLE
add Int support to amd64 and fix issue with structs not being copied

### LINKED ISSUE #289: Unsupported kind int in struct field (amd64 only)
### PureGo Version

main

### Operating System

- [ ] Windows
- [X] macOS
- [ ] Linux
- [ ] FreeBSD
- [ ] Android
- [ ] iOS

### Go Version (`go version`)

go version go1.23.1 darwin/arm64

### What steps will reproduce the problem?

`CGO_ENABLED=1 GOARCH=amd64 go run github.com/hajimehoshi/examples/2048/main.go`

### What is the expected result?

It runs

### What happens instead?

```
2024-10-19 08:29:33.587 ___2go_build_github_com_hajimehoshi_ebiten_v2_examples_2048[27533:680288] [CAMetalLayer nextDrawable] returning nil because allocation failed.
panic: purego: unsupported kind int

goroutine 7 [running]:
github.com/ebitengine/purego.tryPlaceRegister.func2({0x100d586c0?, 0xc0001679c0?, 0xc000156d40?})
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/struct_amd64.go:234 +0x64a
github.com/ebitengine/purego.tryPlaceRegister({0x100d586c0?, 0xc0001679c0?, 0xc000156e00?}, 0xc000081300?, 0xc0002221c0?)
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/struct_amd64.go:247 +0xf6
github.com/ebitengine/purego.addStruct({0x100d586c0?, 0xc0001679c0?, 0x10?}, 0xc0000370a0, 0xc0000370a8, 0xc000037098, 0xc0001572a8, 0xc000157288, 0xc0000372c8, {0x0, ...})
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/struct_amd64.go:103 +0x15f
github.com/ebitengine/purego.addValue({0x100d586c0?, 0xc0001679c0?, 0x2?}, {0x0, 0x0, 0x0}, 0xc0001572a8, 0x0?, 0x0?, 0xc0001570a0, ...)
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/func.go:381 +0x40e
github.com/ebitengine/purego.RegisterFunc.func4({0xc0008ffdb0, 0x3, 0x3})
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/func.go:279 +0xf34
github.com/ebitengine/purego/objc.ID.Send(...)
        /Users/jarrettkuklis/go/pkg/mod/github.com/ebitengine/purego@v0.9.0-alpha.0.20241018101553-999f588e4548/objc/objc_runtime_darwin.go:107
github.com/hajimehoshi/ebiten/v2/internal/graphicsdriver/metal/mtl.RenderCommandEncoder.SetScissorRect(...)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicsdriver/metal/mtl/mtl_darwin.go:916
github.com/hajimehoshi/ebiten/v2/internal/graphicsdriver/metal.(*Graphics).draw(0xc000096000, 0xc00044c0c0, {0xc000219290, 0x1, 0x109b15018?}, {0x0, 0x0, 0x0, 0x0}, 0x0, ...)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicsdriver/metal/graphics_darwin.go:581 +0xe4b
github.com/hajimehoshi/ebiten/v2/internal/graphicsdriver/metal.(*Graphics).DrawTriangles(0xc000096000, 0x400?, {0x0, 0x0, 0x0, 0x0}, 0x3, {0xc000219290, 0x1, 0x1}, ...)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicsdriver/metal/graphics_darwin.go:690 +0x89d
github.com/hajimehoshi/ebiten/v2/internal/graphicscommand.(*drawTrianglesCommand).Exec(0xc00019f5e0, 0xc000832000?, {0x100d962d8, 0xc000096000}, 0x0)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicscommand/command.go:161 +0x1e2
github.com/hajimehoshi/ebiten/v2/internal/graphicscommand.(*commandQueue).flush(0xc00035c630, {0x100d962d8, 0xc000096000}, 0x1, {0x100d90190?, 0x101215d40?})
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicscommand/commandqueue.go:305 +0x5fc
github.com/hajimehoshi/ebiten/v2/internal/graphicscommand.(*commandQueue).Flush.func1()
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicscommand/commandqueue.go:217 +0x78
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Loop.func1({0xc00024a900?, 0xc8?}, 0x0?)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/thread/thread.go:67 +0x4f
github.com/hajimehoshi/ebiten/v2/internal/thread.(*OSThread).Loop(0xc000025000, {0x100d93348, 0xc0008eaf00})
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/thread/thread.go:68 +0x79
github.com/hajimehoshi/ebiten/v2/internal/graphicscommand.LoopRenderThread(...)
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/graphicscommand/thread.go:31
github.com/hajimehoshi/ebiten/v2/internal/ui.(*UserInterface).runMultiThread.func1()
        /Users/jarrettkuklis/Documents/GolandProjects/ebiten/internal/ui/run.go:50 +0x4f
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /Users/jarrettkuklis/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /Users/jarrettkuklis/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x96

Process finished with the exit code 2
```

### Anything else you feel useful to add?

_No response_

### PR DESCRIPTION
**Pull request by TotallyGamerJet**
# What issue is this addressing?
Updates https://github.com/hajimehoshi/ebiten/issues/3135
Closes #289 

## What _type_ of issue is this addressing?
bug fix

## What this PR does | solves
Adds int support on amd64 (arm64 already worked) and fixes a logic bug which stopped structs that should've been copied into memory but weren't (postMerger function)

### CODE REVIEW

**Comment by TotallyGamerJet** [comment]
This fixes the int bug in ebitengine but `BlitCommandEncoder.CopyFromTexture` still needs to use `NSInvocation` because I it has too many parameters for purego but need some more time to research it. This PR is ready to be merged

File: testdata/structtest/struct_test.c [hunk 1]
```diff
@@ -325,3 +332,19 @@ unsigned long InitWithContentRect(int *win, struct Content c, int style, int bac
...
+    return g.a + g.b - g.c + g.d;
+}
+
+struct GoUint4 {
+    GoUint a, b, c, d;
+};
+
+GoUint GoUint4(struct GoUint4 g) {
+    return g.a + g.b + g.c + g.d;
+}
```
Comment (hajimehoshi) [inline testdata/structtest/struct_test.c:350]: Add a new line.

**Review by hajimehoshi — COMMENTED** [review]
Could we file an issue first for PureGo?

**Comment by TotallyGamerJet** [comment]
> Could we file an issue first for PureGo?

done

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

Please add `Closes #289` in the commit comment.

**Pull request merged**